### ISSUE TITLE
Add master volume slider

### ISSUE THREAD

**Issue by yoonthegoon** [wesnoth/wesnoth#7143]
### Describe the desired feature

Add a slider that changes the volume cap.
Useful for when you want to maintain a ratio between the different sound types, but want to decrease the volume for any reason.
Most games ever got this 👍
![image](https://user-images.githubusercontent.com/71526721/202485929-ed310788-6e9d-43bb-9765-f00c16e1fc1f.png)

**Comment by Toranks** [comment]
Also, "Pause music on focus loss" only turns off the music, not the sound effects. The other day I was listening to a bonfire crackle on the PC and I didn't know where it was coming from until I realized it was the game xD
Having a master volume bar would be another solution to this.

**Comment by yoonthegoon** [comment]
Alternatively, adding a `Mute game volume on focus loss` option under `Master volume` would be a nice to have.

**Comment by Wedge009** [comment]
I don't disagree with the proposal - made a quick check to see if there are any other similar open issues, doesn't look it - but I will point out that the desktops I currently use already have a per-application volume control at the OS level. KDE, Windows 7, Windows 10. I imagine such a volume control should link to that per-application control at the OS and not be independent of it.

**Comment by yoonthegoon** [comment]
Yeah i checked all open and closed issues first before I made mine. And I also figured that the OS per-application volume control would come up. But I think it's silly to link it directly to that rather than have an independent control, as that's the way it's handled in most other applications. However, whatever solution that someone comes up with that works, I'll take lol.

I was initially going to work on this a while back, but setting up the dev environment was a big pain for me, not being much of a C++ guy.

**Comment by Wedge009** [comment]
> I think it's silly to link it directly to that rather than have an independent control, as that's the way it's handled in most other applications.

Is that actually a recognised standard? Most applications - where I've bothered to use internal volume controls, anyway - have the 'master' volume linked to the OS volume. Having them independent also seems counter-productive to me, but that could just be a difference of opinion.

**Comment by CelticMinstrel** [comment]
> I don't disagree with the proposal - made a quick check to see if there are any other similar open issues, doesn't look it - but I will point out that the desktops I currently use already have a per-application volume control at the OS level. KDE, Windows 7, Windows 10. I imagine such a volume control should link to that per-application control at the OS and not be independent of it.

I don't think this exists on MacOS? I only have the global volume control and whatever controls an app provides in its own UI. I don't see much of a benefit to adding a master volume slider in Wesnoth though – if I wanted to turn down music and sound volume simultaneously, I'd just adjust the OS volume. And if I was doing it because I wanted to play something in another app, I wouldn't be adjusting the volume in Wesnoth – I'd be muting it.

**Comment by spixi** [comment]
>I currently use already have a per-application volume control at the OS level.

On GNU/Linux this is not done on OS level. There are different applications for sound output. ALSA is the direct way to output sound, but only has one master volume for all applications. That is why you usually use a sound server like PulseAudio or Pipewire. which allows you to individually set volume per application. Wesnoth uses SDL2_mixer which comes with bavkends for ALSA, PulseAudio and Pipewire. As long you have PulseAudio or Pipewire installed, you can use applications like pavucontrol, pamix or certain desktop environment specific applets to control your sound on application level.

**Comment by Pentarctagon** [comment]
That functionality is certainly still available on Linux, regardless of what exactly ends up implementing it.

I agree with CelticMinstrel's previous comment about this not being especially useful as well.
